tests/test_server.py

```diff
@@ -399,16 +401,23 @@ def test_train_nlu_success(
     default_nlu_data: Text,
     default_domain_path: Text,
 ):
-    with ExitStack() as stack:
-        domain_file = stack.enter_context(open(default_domain_path))
-        config_file = stack.enter_context(open(default_stack_config))
-        nlu_file = stack.enter_context(open(default_nlu_data))
+    domain_data = rasa_utils.io.read_yaml_file(default_domain_path)
```

Comment:
How about using `parametrize` or adding another test to make sure it still works with markdown?

Reply:
I took a closer look and there are already a couple of other methods in there that test the same endpoint with markdown files. since this actually trains a model and takes time, I'd rather avoid adding more training runs. what do you think?

Reply:
Yes, but then I'd maybe rename the tests to make it clear that this testing the MD support. Otherwise somebody might come, adapt the test, and suddenly all MD testing is gone.

Another quick way would be to extract the payload extraction to a separate function and test that using `parametrize`.